File: python/structural/flyweight/map/location.py
from map.icon import Icon

class MapLocation:
    def __init__(self, lat: float, long: float, icon: Icon, name: str):
        self.lat = lat
        self.long = long
        self.icon = icon
        self.name = name

    def draw(self):
        print(f"📌 {self.name}")
        Icon.render(self.icon, self.lat, self.long)
        

File: typescript/src/structural/flyweight/map/Location.ts
import { Icon } from "./Icon";

export class MapLocation{
    private lat: number;
    private long: number;
    private icon: Icon;
    private name: string;

    constructor(lat: number, long: number, icon: Icon, name: string){
        this.lat = lat
        this.long = long
        this.icon = icon
        this.name = name
    }

    draw(): void{
        console.log(`📌 ${this.name}`)
        Icon.render(this.icon, this.lat, this.long)
    }
}